bpf/lib/nodeport_egress.h

```diff
@@ -105,7 +105,7 @@ int tail_handle_snat_fwd_ipv6(struct __ctx_buff *ctx)
 
 #ifdef IS_BPF_OVERLAY
 	obs_point = TRACE_TO_OVERLAY;
-#elif defined(IS_BPF_WIREGUARD)
+#elif defined(IS_BPF_WIREGUARD) && IS_BPF_WIREGUARD
```

Comment:
Maybe I'm wrong: `defined(IS_BPF_WIREGUARD) && IS_BPF_WIREGUARD` is always false during clang compiling, so `obs_point = TRACE_TO_CRYPTO` is always compiled out. The further loader injection `consts["IS_BPF_WIREGUARD"] = 1` is too late to bring back the `obs_point = TRACE_TO_CRYPTO` statement.

Did you test and confirm that we can see TRACE_TO_CRYPTO from tail_handle_snat_fwd ? If you tested, then I must be wrong.

Reply:
I thought what you want is to use TRACE_TO_CRYPTO when tail_handle_snat_fwd_ipv6 is called from to_netdev@cilium_wg0? 

If you want to keep TRACE_TO_NETWORK, why do we add:
```c
DEFINE_BOOL(IS_BPF_WIREGUARD, false);
#define IS_BPF_WIREGUARD fetch_bool(IS_BPF_WIREGUARD)
```
to_netdev@cilium_wg0 doesn't have IS_BPF_WIREGUARD in the first place, we already have "when called from bpf_host.c, keep it TRACE_TO_NETWORK".

Reply:
I need that because I'd like TRACE_FROM_CRYPTO in `cil_from_netdev` when attached to Wireguard (ingress).

Reply:
okay I was confused of the direction: only from_netdev will be attached to cilium_wg0. Ignore the nonsense I said above. I believe your code is right, logically.

Instead of defining a `IS_BPF_WIREGUARD` in bpf_host, how about doing it this way:
```diff
- enum trace_point obs_point = IS_BPF_WIREGUARD ? TRACE_FROM_CRYPTO : TRACE_FROM_NETWORK;
+ enum trace_point obs_point = THIS_INTERFACE_IFINDEX == WG_IFINDEX ? TRACE_FROM_CRYPTO : TRACE_FROM_NETWORK;
```

I feel it's easier?

Reply:
Many thanks for the suggestion, let me try this out and I'll back to this soon.
What you're proposing feels way more intuitive: having IS_BPF_WIREGUARD from bpf_host would cause headaches, especially considering that it would be a temporary solution until we move to a new `cil_from_wireguard`.
So yes, I very like `THIS_INTERFACE_IFINDEX == WG_IFINDEX`.